Modules/timemodule.c

```diff
@@ -2074,7 +2076,11 @@ pysleep(_PyTime_t secs)
 #endif
 
     do {
-#ifndef HAVE_CLOCK_NANOSLEEP
+#if defined(HAVE_NANOSLEEP) && !defined(HAVE_CLOCK_NANOSLEEP)
```

Comment:
The change LGTM, but I propose to write the `#ifdef` dance differently and rename `secs` to `timeout` to make the code more readable. I created PR #28545 based on your PR. Is it better? For the main code in Py_BEGIN_ALLOW_THREADS/Py_END_ALLOW_THREADS, I merged the 3 code paths to put if `#ifdef` in the Py_BEGIN_ALLOW_THREADS block. What do you think?